plugins/module_utils/core.py

```diff
@@ -440,6 +440,19 @@ def _boto3_handler(obj):
         return obj
 
 
+def parse_aws_iam_arn(arn):
```

Comment:
Would it be better for this to be more generic.  Rather than limited to just "iam" resources, which I would have said belongs in iam.py, there's probably more of a need for a generic parser, which happens to return IAM resources.

Reply:
Good point; agreed